### PR TITLE
Fix #24 setup process manager with apache airflow

### LINKED ISSUE #24: setup process manager with apache airflow

### PR DESCRIPTION
**Pull request by curioswati**

### CODE REVIEW

**Comment by apoorv74** [comment]
hey @curioswati did you try running this for a sample ? also, can you share the link to the airflow dashboard, we can use that for reporting logs and other metrics

**Comment by curioswati** [comment]
> hey @curioswati did you try running this for a sample ? also, can you share the link to the airflow dashboard, we can use that for reporting logs and other metrics

Yes @apoorv74. The sample run is in progress and the dashboard is live.

File: scheduler/tasks/crawl_treasuries.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+from airflow import DAG
+from airflow.operators.bash_operator import BashOperator
+from airflow.operators.python_operator import BranchPythonOperator
+
+PROJECT_PATH = path.abspath(path.join(path.dirname(__file__), '../..'))
+
+DEFAULT_ARGS = {
+    'owner': 'airflow',
+    'start_date': dt.datetime(2019, 8, 1),
```
Comment (gggodhwani) [inline scheduler/tasks/crawl_treasuries.py]: Can we please set it to the first of the month whenever the call happen?

File: scheduler/tasks/crawl_treasuries.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    'concurrency': 1,
+    'retries': 0
```
Comment (gggodhwani) [inline scheduler/tasks/crawl_treasuries.py:22]: Can we explain the reason of keeping the retry count keeping 0?

File: scheduler/tasks/crawl_treasuries.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    # check if the execution day is 'Friday'
+    if execution_date.weekday() == 3:
+        return ['crawl_ddo_codes', 'crawl_expenditure', 'crawl_receipts']
+
+    return ['crawl_expenditure', 'crawl_receipts']
+
+
+with DAG('crawl_treasuries',
+         default_args=DEFAULT_ARGS,
+         schedule_interval='30 4 * * *',
```
Comment (gggodhwani) [inline scheduler/tasks/crawl_treasuries.py]: Can we explain the CRON equivalent & timezone?

File: scheduler/tasks/crawl_treasuries.py [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+        )
+    )
+
+    CRAWL_DDO_CODES = BashOperator(
+        task_id='crawl_ddo_codes',
+        bash_command='cd {}/scraper && scrapy crawl ddo_collector'.format(PROJECT_PATH)
+    )
+
+    EXP_CRAWL_COMMAND = Template("""
+        cd $project_path/scraper && scrapy crawl treasury_expenditures -a start={{ ds_nodash }} -a end={{ ds_nodash }}
```
Comment (gggodhwani) [inline scheduler/tasks/crawl_treasuries.py:56]: Can we please link the Airflow docs here to explain Jinja variables if any?

File: scheduler/tasks/crawl_treasuries.py [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    """)
+
+    CRAWL_EXPENDITURE = BashOperator(
```
Comment (gggodhwani) [inline scheduler/tasks/crawl_treasuries.py]: Can we make the nomenclature consistent, something like: EXP_CRAWL_TASK

File: scheduler/tasks/crawl_treasuries.py [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+        task_id='crawl_expenditure',
+        bash_command=EXP_CRAWL_COMMAND.substitute(project_path=PROJECT_PATH),
+        trigger_rule='none_failed'
+    )
+
+    REC_CRAWL_COMMAND = Template("""
+        cd $project_path/scraper && scrapy crawl treasury_receipts -a start={{ ds_nodash }} -a end={{ ds_nodash }}
+    """)
+
+    CRAWL_RECEIPTS = BashOperator(
```
Comment (gggodhwani) [inline scheduler/tasks/crawl_treasuries.py]: Same nomenclature thing

File: scraper/scraper/spiders/treasury_base.py [hunk 7]
```diff
@@ -127,6 +132,8 @@ def parse(self, response):
                 }
                 yield self.make_dataset_request(params)
 
+            time.sleep(5)
```
Comment (gggodhwani) [inline scraper/scraper/spiders/treasury_base.py:136]: Comment: # add a sleep of 5 seconds after each treasury crawled

**Review by gggodhwani — CHANGES_REQUESTED** [review]
Requesting some minor changes

**Review by gggodhwani — APPROVED** [review]
:+1:

**Pull request merged**